include/triton/Analysis/AxisInfo.h

```diff
@@ -202,8 +202,11 @@ class ModuleAxisInfoAnalysis : public CallGraph<AxisInfoMapT> {
     return &(it->second);
   }
 
-  unsigned getPtrContiguity(Value ptr);
-  unsigned getPtrAlignment(Value ptr);
+  // Allow to override the type of the ptr to avoid recreating AxisAnalysis when
```

Comment:
This might be a minor thing but to me it's better to have separate methods so we can better separate and document the use cases here. Basically have the following

```c++

unsigned getPtrContiguity(Value ptr);
unsigned getPtrAlignment(Value ptr);

// Overloads of the above methods but have separated base scalar pointers
// and the tensor type to read from it. These are useful for computing axis info
// when lowering to hardware intrinsics that require separate scalar/warp-uniform
// vs tensor/thread-specific indexing, e.g., AMD buffer intrinsics.
unsigned getPtrContiguity(Value scalarPtr, RankedTensorType tensorType);
unsigned getPtrAlignment(Value scalarPtr, RankedTensorType tensorType);
```

Reply:
I opted to make the function more general as suggested by @giuseros as it was only used in a couple of places. The name was already a bit misleading as it had a check for non tensor types and returned 1 so you could call it with any Value not only ptrs.
If this is not fine I can split them again :).

Reply:
okay this is fine for me then. also @Jokeren here in case there are concerns.

Reply:
IIUC, nothing was wrong with the nvidia backend, right? I'm OK with renaming

Reply:
Correct--nothing wrong / needed changing on nvidia side. This is needed for AMD path to lower to buffer intrinsics, where we need separate scalar/warp-uniform vs tensor/thread-specific indexing. So instead of `tensor<128x64x!tt.ptr<f16>>`, we need to do the analysis on a `!tt.ptr<f16>` base scalar pointer + a `tensor<128x64xi32>` offset.